include/libdnf/base/transaction_package.hpp

```diff
@@ -25,7 +25,6 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 #include "libdnf/rpm/package.hpp"
 #include "libdnf/transaction/transaction_item_action.hpp"
```

Comment:
Please remove the prefix in commit heasder

Reply:
I am sorry but I somehow feel that the removal is not a good idea. It means the removal needs a discussion with the way, how commands after transaction will have an access to that information or how the information will be propagated into history DB

Reply:
User does not have an access to rpm transaction packages but only to libdnf::base::Transaction class and transaction packages stored inside of class. Transaction class runs the transaction therefore it is quite logical to have stored a state of transaction items in packages provided by Transaction class. And I also think that we can add additional state that will represent situation not started yet.

Reply:
I've had a look at this and there's likely only one way to do this that would make sense, and that is updating the state in the rpm transaction callback. Otherwise, we could only bulk-update all transaction items to either `OK` or `ERROR` depending on the transaction result, but that wouldn't really be useful for anything.

But updating from rpm transaction callback seems to be the way to go, the `TransactionPackage` is already stored as the "data" of the callback, so it should be easy to do and we could even introduce finer state granularity, as you say "not started" and we could also have something like "rpm installed", "scriptlets run" and "finished". That could actually be quite helpful in recovering from unfinished transactions. We could store the same in transaction history, right now the state in transaction history is not being updated at all.

I'll drop the commit removing the state from `TransactionPackage` from the PR and properly setting the state (along with defining the values) will be followup material.